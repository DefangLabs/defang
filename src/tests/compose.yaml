name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
