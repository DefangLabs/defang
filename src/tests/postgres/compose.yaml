services:
  x:
    image: postgres
    x-defang-postgres:

  y:
    image: example
    x-defang-postgres:

  z:
    image: postgres
