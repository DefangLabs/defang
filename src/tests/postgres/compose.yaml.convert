[
  {
    "name": "x",
    "image": "postgres",
    "platform": 2,
    "internal": true,
    "networks": 1,
    "postgres": {}
  },
  {
    "name": "y",
    "image": "example",
    "platform": 2,
    "internal": true,
    "networks": 1,
    "postgres": {}
  },
  {
    "name": "z",
    "image": "postgres",
    "platform": 2,
    "internal": true,
    "networks": 1
  }
]