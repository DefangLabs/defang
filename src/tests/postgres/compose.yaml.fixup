[
  {
    "command": null,
    "entrypoint": null,
    "image": "postgres",
    "networks": {
      "default": null
    }
  },
  {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "default": null
    }
  },
  {
    "command": null,
    "entrypoint": null,
    "image": "postgres",
    "networks": {
      "default": null
    }
  }
]