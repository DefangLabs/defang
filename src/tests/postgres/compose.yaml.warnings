 ! service "x": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "x": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "y": managed Postgres service should use a postgres image
 ! service "y": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "y": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "z": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "z": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "z": stateful service will lose data on restart; use a managed service instead