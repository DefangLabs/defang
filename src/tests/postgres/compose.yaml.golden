name: postgres
services:
  x:
    image: postgres
    networks:
      default: null
    x-defang-postgres: null
  "y":
    image: example
    networks:
      default: null
    x-defang-postgres: null
  z:
    image: postgres
    networks:
      default: null
networks:
  default:
    name: postgres_default
